iree/compiler/Codegen/Common/ConvertToDestinationPassingStylePass.cpp

```diff
@@ -324,6 +323,26 @@ SmallVector<NamedAttribute> PruneAttributeList(linalg::GenericOp op) {
   return preservedAttrs;
 }
 
+static bool isFromReadOnlyTensor(Value v) {
+  return TypeSwitch<Operation *, bool>(v.getDefiningOp())
+      .Case<arith::ConstantOp>(
+          [&](arith::ConstantOp constantOp) { return true; })
+      .Case<tensor::CollapseShapeOp, tensor::ExpandShapeOp>(
+          [&](auto op) { return isFromReadOnlyTensor(op.src()); })
+      .Case<tensor::ExtractSliceOp>([&](tensor::ExtractSliceOp sliceOp) {
```

Comment:
You probably also want to add `tensor::CastOp` also in there. Those will be there for dynamic shape cases.